packages/formspree-react/src/context.tsx

```diff
@@ -33,65 +26,40 @@ const getStripe = (stripeKey: string) => {
   return stripePromise;
 };
 
-const handleCreateClient = (promise?: Stripe, project?: string) => {
-  const config: Config = {};
-
-  if (promise) {
-    config.stripePromise = promise;
-  }
-
-  if (project) {
-    config.project = project;
-  }
-
-  return createClient(config);
-};
-
-export const FormspreeProvider = (props: FormspreeProviderProps) => {
-  const [stateStripePromise, setStateStripePromise] = useState<
-    Stripe | undefined
-  >(undefined);
-  const [client, setClient] = useState<Client>(
-    handleCreateClient(stateStripePromise, props.project)
-  );
+/**
+ * FormspreeProvider creates Formspree Client based on the given props
+ * and makes the client available through via context.
+ */
+export function FormspreeProvider(props: FormspreeProviderProps) {
+  const { children, project, stripePK } = props;
+  const [client, setClient] = useState<Client>(createClient({ project }));
 
   useEffect(() => {
-    const getStripePromise = async (stripeKey: string) => {
-      const promiseStripe = await getStripe(stripeKey);
-      if (promiseStripe) {
-        setStateStripePromise(promiseStripe);
-      }
-    };
-
-    if (props.stripePK) {
-      getStripePromise(props.stripePK);
+    let isMounted = true;
+    if (stripePK) {
+      getStripe(stripePK).then((stripe) => {
+        if (stripe && isMounted) {
+          setClient((client) => createClient({ ...client, stripe }));
+        }
+      });
     }
-  }, [props.stripePK]);
+    return () => {
+      isMounted = false;
+    };
+  }, [stripePK]);
 
   useEffect(() => {
-    if (stateStripePromise) {
-      setClient(handleCreateClient(stateStripePromise, props.project));
-    }
-  }, [props.project, stateStripePromise]);
+    setClient((client) =>
+      client.project !== project ? createClient({ ...client, project }) : client
+    );
+  }, [project]);
 
   return (
     <FormspreeContext.Provider value={{ client }}>
-      {props.stripePK ? (
-        <>
-          {stateStripePromise && (
-            <Suspense fallback={<p>....</p>}>
-              <Elements stripe={stateStripePromise}>
-                <>{props.children}</>
-              </Elements>
-            </Suspense>
-          )}
-        </>
-      ) : (
-        <>{props.children}</>
-      )}
+      {children}
```

Comment:
I looked through `Elements` (provider) from `@stripe/react-stripe-js@1.16.5`; it pretty much is a provider to pass down `stripe` and `elements` (get from `stripe.elements()`), and cart context that we don't need.

We have everything that we need in `client.stripe`, so this changes so we get it `client.stripe` instead. It'll prevent the Stripe client being different on `client.stripe` vs `stripe` from `ElementsContext`.

This also eliminates the need to call `useStripe` and `useElements` conditionally (breaking Rules of Hooks). See https://github.com/formspree/formspree-js/pull/48/files#diff-c2202ad14f7a0077d1c91053ad42448dd7078b3f5fa941c52238ccc43d338128L69-L76

Reply:
I thought the point of this conditional loading, along with the Suspense and lazy() library load, was to ensure that the stripe script only loads if a stripePK is provided.

In other words, before we added this lazy loader, people who viewed inspector would see a network call to stripe.js. Adding this lazy loader avoided the `<script src="https://js.stripe.com/v3">` from being injected.

See https://github.com/formspree/formspree-js/issues/16

Can we test to ensure we're not reverting back to that previous behavior of loading stripe.js even when no stripePK is configured?

Reply:
It is good to double check. The lazy loading of Stripe script is handled here: https://github.com/formspree/formspree-js/pull/48/files#diff-acd46b99f23f96931c115e5f3dd6159a1a4ae7fcb4421cc3c936d9647a1a7e05R39-R40. We don't load stripe if `stripePK` isn't provided. We can see from the [preview app](https://formspree-react-demo-6uz98uo9c-formspree.vercel.app/) on Simple Form that stripe is not loaded.

<img width="1416" alt="image" src="https://github.com/formspree/formspree-js/assets/3942264/9f5e2c84-bded-4841-b241-8748786ce7a4">

The lazy loading of the `Element` has nothing to do with the Stripe script but to split the chunk containing `@stripe/react-stripe-js` and load it on demand, however, in the previous implementation, `@stripe/react-stripe-js` is loaded anyway because we import it synchronously in `useForm`: https://github.com/formspree/formspree-js/blob/main/packages/formspree-react/src/useForm.ts#L4C2-L4C2

However, I found that the manual context passing breaks the preview app when it renders `CardElement` since [it reads from `ElementsContext` under the hood](https://github.com/stripe/react-stripe-js/blob/809f84bc4ca11920ea89e6fb06312499ecb45f71/src/components/createElementComponent.tsx#L72), so we will have to revert back to pass the context down via `Elements` provider.